gruppi/viste.py

```diff
@@ -66,6 +136,25 @@ def attivita_gruppi_gruppo_abbandona(request, me, pk):
                               torna_url="/attivita/gruppi/")
 
 
+@pagina_privata
+def attivita_gruppi_gruppo_elimina(request, me, pk):
+    """
+    Elimina un gruppo.
+    """
+    # Verifica se sono un delegato d'area.
+    if not me.deleghe_attuali(tipo=DELEGATO_AREA).exists():
```

Comment:
⚠️ 🔐 Se sono un delegato di una qualsiasi area in Gaia posso cancellare *tutti* i gruppi in Gaia?

Mi sa che questo vuol dire che abbiamo bisogno di un nuovo permesso. Fai riferimento alla guida per la creazione di deleghe e permessi, e aggiungi dei nuovi permessi (qualcosa come `GESTIONE_GRUPPO`) ai delegati di area e ai referenti di attivita'.

https://github.com/CroceRossaItaliana/jorvik/wiki/Deleghe,-Permessi-e-Livelli-di-Accesso#guide-creazione-deleghe-e-permessi

Questo ti permettera' poi di usare il sistema per il controllo dei permessi, per controllare se l'utente puo' modificare il gruppo individuale:
```python
if not me.permessi_almeno(gruppo, MODIFICA):
    return redirect(ERRORE_PERMESSI)
```

✨

Reply:
@AlfioEmanueleFresta potrebbe andar bene anche questa soluzione?
In pratica si verifica se la sede del gruppo è tra quelle in cui sono delegato d'area.

    # Verifica se sono un delegato d'area.
    delegato_area = me.deleghe_attuali(tipo=DELEGATO_AREA)
    if not delegato_area.exists():
        return redirect(ERRORE_PERMESSI)

    gruppo = get_object_or_404(Gruppo, pk=pk)

    # Ottieni le sedi in cui sono delegato d'area e vedi se la sede del gruppo è tra queste.
    delega_gruppo_sede = me.sedi_deleghe_attuali(deleghe=delegato_area).filter(id=gruppo.sede.id)

    # Se il Queryset è vuoto non ho i permessi per eliminare il gruppo.
    if not delega_gruppo_sede.exists():
        return redirect(ERRORE_PERMESSI)

    gruppo.delete()